sycl/include/sycl/ext/oneapi/experimental/backend/backend_traits_cuda.hpp

```diff
@@ -0,0 +1,165 @@
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the specializations of the sycl::detail::interop,
+// sycl::detail::BackendInput and sycl::detail::BackendReturn class templates
+// for the CUDA backend but there is no sycl::detail::InteropFeatureSupportMap
+// specialization for the CUDA backend.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/accessor.hpp>
+#include <CL/sycl/context.hpp>
+#include <CL/sycl/detail/backend_traits.hpp>
+#include <CL/sycl/device.hpp>
+#include <CL/sycl/event.hpp>
+#include <CL/sycl/kernel_bundle.hpp>
+#include <CL/sycl/queue.hpp>
+
+#include <vector>
+
+typedef int CUdevice;
+typedef struct CUctx_st *CUcontext;
+typedef struct CUstream_st *CUstream;
+typedef struct CUevent_st *CUevent;
+typedef struct CUmod_st *CUmodule;
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+// TODO the interops for context, device, event, platform and program
+// may be removed after removing the deprecated 'get_native()' methods
+// from the corresponding classes. The interop<backend, queue> specialization
+// is also used in the get_queue() method of the deprecated class
+// interop_handler and also can be removed after API cleanup.
+template <> struct interop<backend::ext_oneapi_cuda, context> {
+  using type = CUcontext;
+};
+
+template <> struct interop<backend::ext_oneapi_cuda, device> {
+  using type = CUdevice;
+};
+
+template <> struct interop<backend::ext_oneapi_cuda, event> {
+  using type = CUevent;
+};
+
+template <> struct interop<backend::ext_oneapi_cuda, queue> {
+  using type = CUstream;
+};
+
+template <> struct interop<backend::ext_oneapi_cuda, platform> {
+  using type = std::vector<CUdevice>;
+};
+
+#ifdef __SYCL_INTERNAL_API
+template <> struct interop<backend::ext_oneapi_cuda, program> {
+  using type = CUmodule;
+};
+#endif
+
+template <typename DataT, int Dimensions, access::mode AccessMode>
+struct interop<backend::ext_oneapi_cuda,
+               accessor<DataT, Dimensions, AccessMode, access::target::device,
+                        access::placeholder::false_t>> {
+  using type = DataT *;
+};
+
+template <typename DataT, int Dimensions, access::mode AccessMode>
+struct interop<
+    backend::ext_oneapi_cuda,
+    accessor<DataT, Dimensions, AccessMode, access::target::constant_buffer,
+             access::placeholder::false_t>> {
+  using type = DataT *;
+};
+
+template <typename DataT, int Dimensions, access::mode AccessMode>
+struct interop<backend::ext_oneapi_cuda,
+               accessor<DataT, Dimensions, AccessMode, access::target::local,
+                        access::placeholder::false_t>> {
+  using type = DataT *;
+};
```

Comment:
What are these used for? I don't believe accessor has `make_*` or `get_native*` variants.

Note: `get_native_mem` takes an accessor, but its return type is `backend_return_t<Backend, buffer<DataT, Dims>>`.

Note 2: Even if the CUDA backend defines an interface for it, is it ever safe to have a input/output type for `access::target::local`?

Reply:
These are used for kernel function interop. Further changes are needed to enable this functionality. I will remove these for the time being and re-introduce them later.

Reply:
By "kernel function interop", do you mean interop inside a kernel? If so, couldn't users simply use the `get_pointer()` member function of the accessor?

Reply:
That is a good question. I am looking into it further. I have removed reference to `get_native` for accessors in this PR till it is clear what needs to happen.